actionpack/lib/action_controller/railtie.rb

```diff
@@ -67,5 +67,12 @@ class Railtie < Rails::Railtie #:nodoc:
         config.compile_methods! if config.respond_to?(:compile_methods!)
       end
     end
+
+    initializer "setup session store" do |app|
+      ActiveSupport.on_load(:action_controller) do
```

Comment:
We have lot of tests which set the session_store in `config/application.rb`. If this is not in the inside of on_load hook, then it gets run after the application.rb is run -- overriding the config set in application.rb.